scvi/lightning/_trainingplans.py

```diff
@@ -571,3 +571,85 @@ def optimizer_step(self, *args, **kwargs):
 
     def backward(self, *args, **kwargs):
         pass
+
+
+class ClassifierTrainingPlan(pl.LightningModule):
+    """
+    Lightning module task to train a simple MLP classifier.
+
+    Parameters
+    ----------
+    classifier
+        A model instance from :class:`~scvi.modules.Classifier`.
+    lr
+        Learning rate used for optimization.
+    weight_decay
+        Weight decay used in optimizatoin.
+    eps
+        eps used for optimization.
+    optimizer
+        One of "Adam" (:class:`~torch.optim.Adam`), "AdamW" (:class:`~torch.optim.AdamW`).
+    data_key
+        Key for classifier input in tensor dict minibatch
+    labels_key
+        Key for classifier label in tensor dict minibatch
+    loss
+        PyTorch loss to use
+    """
+
+    def __init__(
+        self,
+        classifier: BaseModuleClass,
+        lr: float = 1e-3,
+        weight_decay: float = 1e-6,
+        eps: float = 0.01,
+        optimizer: Literal["Adam", "AdamW"] = "Adam",
+        data_key: str = _CONSTANTS.X_KEY,
+        labels_key: str = _CONSTANTS.LABELS_KEY,
+        loss: Callable = torch.nn.CrossEntropyLoss,
+    ):
+        super(TrainingPlan, self).__init__()
```

Comment:
@adamgayoso should this be pl.LightningModule or should ClassifierTrainingPlan inherit TrainingPlan?


Ah I think PyroTrainingPlan is the winning ticket

Reply:
Ah I think `PyroTrainingPlan` is the winning ticket

Reply:
maybe not though...

Reply:
whoops.... you actually shouldn't need to specify anything
```suggestion
        super().__init__()
```
but it should be `pl.LightningModule`